src/simularium/VisGeometry.ts

```diff
@@ -843,13 +873,45 @@ class VisGeometry {
     }
 
     private attemptToLoadGeometry(
-        url: string,
+        urlOrPath: string,
         registry: Map<string | number, PDBModel | MeshLoadRequest>,
-        loadFunctionName: string
+        displayType: GeometryDisplayType
     ) {
-        if (!registry.has(url) && !this.geoLoadAttempted.get(url)) {
-            this.geoLoadAttempted.set(url, true);
-            return this[loadFunctionName](url);
+        if (this.localGeoFiles.has(urlOrPath)) {
```

Comment:
should this line check in the registry also, like line 898?  if item with the given key is in the registry we can skip all this?

Reply:
Since I delete it from the cache on line 896, after it's been loaded in the registry, the assumption is that it's only in the cache if it's never been loaded before.